### ISSUE TITLE
Header in 307 answer is not CRLF but just LF

### ISSUE THREAD

**Issue by flaf** [e2guardian/e2guardian#288]
Hi,

After a DENIED on a HTTP*S* page, the answer 307 of e2guardian is not CRLF but just LF:

```
~$ export https_proxy='http://enzo.canot:password@192.168.226.201:8080'

~$ curl -s https://www.playboy.com -I
HTTP/1.1 307 Temporary Redirect
Location: https://edu-id.ac-versailles.fr/denyssl.php?DENIEDURL==https%3a%2f%2fwww%2eplayboy%2ecom::IP==172.31.130.19::USER==enzo.canot::CATEGORIES==::REASON==Site%20interdit%20%3a%20playboy%2ecom
Content-Length: 0
Cache-control: no-cache
Connection: close

~$ curl -s https://www.playboy.com -I | od -c
0000000   H   T   T   P   /   1   .   1       3   0   7       T   e   m
0000020   p   o   r   a   r   y       R   e   d   i   r   e   c   t  \n
0000040   L   o   c   a   t   i   o   n   :       h   t   t   p   s   :
0000060   /   /   e   d   u   -   i   d   .   a   c   -   v   e   r   s
0000100   a   i   l   l   e   s   .   f   r   /   d   e   n   y   s   s
0000120   l   .   p   h   p   ?   D   E   N   I   E   D   U   R   L   =
0000140   =   h   t   t   p   s   %   3   a   %   2   f   %   2   f   w
0000160   w   w   %   2   e   p   l   a   y   b   o   y   %   2   e   c
0000200   o   m   :   :   I   P   =   =   1   7   2   .   3   1   .   1
0000220   3   0   .   1   9   :   :   U   S   E   R   =   =   e   n   z
0000240   o   .   c   a   n   o   t   :   :   C   A   T   E   G   O   R
0000260   I   E   S   =   =   :   :   R   E   A   S   O   N   =   =   S
0000300   i   t   e   %   2   0   i   n   t   e   r   d   i   t   %   2
0000320   0   %   3   a   %   2   0   p   l   a   y   b   o   y   %   2
0000340   e   c   o   m  \n   C   o   n   t   e   n   t   -   L   e   n
0000360   g   t   h   :       0  \n   C   a   c   h   e   -   c   o   n
0000400   t   r   o   l   :       n   o   -   c   a   c   h   e  \n   C
0000420   o   n   n   e   c   t   i   o   n   :       c   l   o   s   e
0000440  \n
0000441
```

It happens only with 307 answers. For instance, no problem with 407 answer:

```
~$ export https_proxy='http://192.168.226.201:8080'
~$ curl -s https://www.playboy.com -I | od -c
0000000   H   T   T   P   /   1   .   1       4   0   7       P   r   o
0000020   x   y       A   u   t   h   e   n   t   i   c   a   t   i   o
0000040   n       R   e   q   u   i   r   e   d  \r  \n   S   e   r   v
0000060   e   r   :       s   q   u   i   d   /   3   .   4   .   8  \r
0000100  \n   M   i   m   e   -   V   e   r   s   i   o   n   :       1
0000120   .   0  \r  \n   D   a   t   e   :       M   o   n   ,       2
0000140   3       O   c   t       2   0   1   7       1   5   :   1   8
0000160   :   5   9       G   M   T  \r  \n   C   o   n   t   e   n   t
0000200   -   T   y   p   e   :       t   e   x   t   /   h   t   m   l
0000220  \r  \n   C   o   n   t   e   n   t   -   L   e   n   g   t   h
0000240   :       3   1   5   5  \r  \n   X   -   S   q   u   i   d   -
0000260   E   r   r   o   r   :       E   R   R   _   C   A   C   H   E
0000300   _   A   C   C   E   S   S   _   D   E   N   I   E   D       0
0000320  \r  \n   P   r   o   x   y   -   A   u   t   h   e   n   t   i
0000340   c   a   t   e   :       B   a   s   i   c       r   e   a   l
0000360   m   =   "   "   P   r   o   x   y       A   c   a   d   e   m
0000400   i   q   u   e   "   "  \r  \n   X   -   C   a   c   h   e   :
0000420       M   I   S   S       f   r   o   m       f   i   l   t   r
0000440   a   g   e   -   a   c   a   d   e   m   i   q   u   e  \r  \n
0000460   X   -   C   a   c   h   e   -   L   o   o   k   u   p   :    
0000500   N   O   N   E       f   r   o   m       f   i   l   t   r   a
0000520   g   e   -   a   c   a   d   e   m   i   q   u   e   :   3   1
0000540   2   8  \r  \n   V   i   a   :       1   .   1       f   i   l
0000560   t   r   a   g   e   -   a   c   a   d   e   m   i   q   u   e
0000600       (   s   q   u   i   d   /   3   .   4   .   8   )  \r  \n
0000620   C   o   n   n   e   c   t   i   o   n   :       c   l   o   s
0000640   e  \r  \n  \r  \n
0000645
```

The absence of CRLF is a problem for me because I have a load balancer (F5) which drops the HTTP packets with LF-header. Furthermore, I think CRLF is highly recommended.

Regards.

**Comment by fredbcode** [comment]
Which version ?

I guess you are using sslaccedeniedadress, right ?
If yes you can make a try to change src/ConnecHandler.cpp 4112 and compile (if you know exactly what you are doing)
 




----- Mail original -----

**Comment by flaf** [comment]
Sorry, it is the latest version 4.1.3. I have used the .deb package.

**Comment by flaf** [comment]
And yes in the conf I have:

```
sslaccessdeniedaddress = 'https://edu-id.ac-versailles.fr/denyssl.php'
```

**Comment by fredbcode** [comment]
Fixed on 4.1.4 check needed on v5 before close

**Comment by fredbcode** [comment]
ok done
